collection.go

```diff
@@ -185,33 +192,60 @@ func (c *collection) Find(query Query) ([]Document, error) {
 	var docs []Document
 
 	i := c.findIndex(query)
-
 	if i == nil {
-		return nil, errors.New("no index found")
+		return nil, ErrNoIndex
 	}
 
-	err := c.db.View(func(tx *bbolt.Tx) error {
-		// nil is not possible since adding an index creates the iBucket
-		iBucket := tx.Bucket([]byte(c.Name))
-
-		refs, err := i.Find(iBucket, query)
+	// the iteratorFn that collects results in a slice
+	refMap := map[string]bool{}
+	var newRef = make([]Reference, 0)
+	var refFn = func(key []byte, value []byte) error {
+		refs, err := entryToSlice(value)
 		if err != nil {
 			return err
 		}
+		for _, r := range refs {
+			if _, b := refMap[r.EncodeToString()]; !b {
```

Comment:
can everything that can be expressed in bytes be safely represented as string (without collisions)?

I think you can index on an array (only works if the lengths are fixed though)

Reply:
they are `references`, a custom type which does hex encoding of byte arrays.